src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -566,12 +621,120 @@ public void testCreateVCFWriterDefaults(
         Assert.assertFalse(outFileMD5.exists(), "An md5 file was created and should not have been");
     }
 
-    private TestGATKToolWithVariants createTestVariantTool(final String args[]) {
-        final TestGATKToolWithVariants tool = new TestGATKToolWithVariants();
-        if (null != args) {
-            final CommandLineParser clp = new CommandLineArgumentParser(tool);
-            clp.parseArguments(System.out, args);
+    private List<Annotation> instantiateAnnotations(final CommandLineParser clp) {
+        GATKAnnotationPluginDescriptor annotationPlugin = clp.getPluginDescriptor(GATKAnnotationPluginDescriptor.class);
+        return annotationPlugin.getResolvedInstances();
+    }
+
+    @Test
+    public void testMakeEmptyAnnotations() {
+        final TestGATKToolWithVariants tool = createTestVariantTool(null);
+        Collection<Annotation> annots = tool.makeVariantAnnotations();
+
+        Assert.assertTrue(annots.isEmpty());
+    }
+
+    @Test
+    public void testGetAllAnnotations() {
+        String[] args = {"--"+StandardArgumentDefinitions.ENABLE_ALL_ANNOTATIONS};
+
+        final TestGATKToolWithVariants tool = createTestVariantTool(args);
+        Collection<Annotation> annots = tool.makeVariantAnnotations();
+
+        ClassFinder finder = new ClassFinder();
+        finder.find(GATKAnnotationPluginDescriptor.pluginPackageName, Annotation.class);
+
+        Set<Class<?>> classes = finder.getConcreteClasses();
+        Assert.assertFalse(classes.isEmpty());
+        Assert.assertEquals(annots.size(),classes.size());
+        for(Class<?> found : classes) {
+            Assert.assertTrue(annots.stream().anyMatch(a -> a.getClass()==found));
         }
+    }
+
+    @Test
+    public void testExcludeAnnotation(){
+        String[] args = {"--"+StandardArgumentDefinitions.ENABLE_ALL_ANNOTATIONS, "-AX", "Coverage"};
+
+        final TestGATKToolWithVariants tool = createTestVariantTool(args);
+        Collection<Annotation> annots = tool.makeVariantAnnotations();
+
+        // Asserting that the annotation was excluded
+        Assert.assertFalse(annots.stream().anyMatch(a -> a.getClass()==Coverage.class));
+    }
+
+    @Test
+    public void testIncludeGroupAnnotations(){
+        String[] args = {"-G", StandardAnnotation.class.getSimpleName()};
+
+        final TestGATKToolWithVariants tool = createTestVariantTool(args);
+        Collection<Annotation> annots = tool.makeVariantAnnotations();
+
+        // Asserting that a standard annotation was included but not everything
+        Assert.assertTrue(annots.stream().anyMatch(a -> a.getClass()==Coverage.class));
+        Assert.assertFalse(annots.stream().anyMatch(a -> a.getClass()==StandardAnnotation.class));
+        Assert.assertFalse(annots.stream().anyMatch(a -> a.getClass()==ClippingRankSumTest.class));
```

Comment:
Use a `ClassFinder` to find all concrete `StandardAnnotations`, and assert that you get them all.